nix/hevm-tests/default.nix

```diff
@@ -1,6 +1,6 @@
 { pkgs }:
 let
-  solc = "${pkgs.solc-versions.solc_0_6_7}/bin/solc";
+  solc = "${pkgs.solc-static-versions.solc_0_8_6}/bin/solc-0.8.6";
   solidity = pkgs.fetchFromGitHub {
```

Comment:
would be nice to bump this at some point as well to get the latest smtchecker tests, but I left it for now since it's probably a pretty big job and this PR was already fairly time intensive...